src/Stratis.Bitcoin.Features.Api/MvcBuilderExtensions.cs

```diff
@@ -16,18 +17,15 @@ public static class MvcBuilderExtensions
         /// <returns>The Mvc builder</returns>
         public static IMvcBuilder AddControllers(this IMvcBuilder builder, IServiceCollection services)
         {
-            // Adds Controllers with API endpoints
-            System.Collections.Generic.IEnumerable<ServiceDescriptor> controllerTypes = services.Where(s => s.ServiceType.GetTypeInfo().BaseType == typeof(Controller));
-            foreach (ServiceDescriptor controllerType in controllerTypes)
-            {
-                builder.AddApplicationPart(controllerType.ServiceType.GetTypeInfo().Assembly);
-            }
+            // Adds the assembles containing our features.
+            var featureAssemblies = AppDomain.CurrentDomain.GetAssemblies().SelectMany(x => x.GetTypes())
+                .Where(x => typeof(FeatureController).IsAssignableFrom(x) && !x.IsInterface && !x.IsAbstract)
+                .Select(x => x.Assembly)
+                .ToList();
```

Comment:
This will resolve all feature assemblies, even if the feature isn't explicitly enabled in the node builder. Do we want this?

Reply:
Good spot. Refactored to include all features assemblies of selected features only.